### PR TITLE
Add Makefile to initialise tests to make sure docker is running.

### PR DESCRIPTION
**Pull request by Nyholm**
After this PR, nothing stops me from actually writing a test =)

### CODE REVIEW

File: src/S3/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.EXPORT_ALL_VARIABLES:
+
+# Build the PHP runtimes
+initialize:
+	docker pull lphoward/fake-s3
+	docker start async_aws_s3 || docker run -d -p 4569:4569 --name async_aws_s3 lphoward/fake-s3
```
Comment (jderusse) [inline src/S3/Makefile]: `docker rm -fv` to clear previous test ?

Comment (Nyholm) [reply src/S3/Makefile]: I should probably add a "stop" method. But I want to "start all" so I can run my test one by one..

**Comment by Nyholm** [comment]
The test fails because a test in S3 shows a bug in Core. 

That bug is fixed in this PR too.

File: src/Core/StreamableBody.php [hunk 2]
```diff
@@ -43,24 +43,23 @@ public function getContentAsString(): string
     {
         $resource = $this->getContentAsResource();
 
-        return \stream_get_contents($resource);
+        try {
+            return \stream_get_contents($resource);
+        } finally {
+            \fclose($resource);
```
Comment (jderusse) [inline src/Core/StreamableBody.php]: good catch

Comment (Nyholm) [reply src/Core/StreamableBody.php]: Yeah. I guess it is a good idea to have tests after all =)

File: src/Core/StreamableBody.php [hunk 3]
```diff
@@ -43,24 +43,23 @@ public function getContentAsString(): string
...
+        }
     }
 
     public function getContentAsResource()
     {
         $resource = \fopen('php://temp', 'rw+');
```
Comment (jderusse) [inline src/Core/StreamableBody.php]: shouldn't it be closed if method failed?

Comment (Nyholm) [reply src/Core/StreamableBody.php]: Yeah, Good

**Pull request merged**